cg/store/api/findbasicdata.py

```diff
@@ -128,13 +133,17 @@ def latest_version(self, tag: str) -> Optional[ApplicationVersion]:
             application_obj.versions[-1] if application_obj and application_obj.versions else None
         )
 
-    def organism(self, internal_id: str) -> Organism:
+    def get_organism_by_internal_id(self, internal_id: str) -> Organism:
         """Find an Organism by internal_id."""
-        return self.Organism.query.filter_by(internal_id=internal_id).first()
+        return apply_organism_filter(
+            organisms=self._get_organism_query(),
+            filters=[OrganismFilter.FILTER_BY_INTERNAL_ID],
+            internal_id=internal_id,
+        ).first()
 
     def organisms(self) -> List[Organism]:
         """Fetch all organisms"""
-        return self.Organism.query.order_by(Organism.internal_id)
+        return self._get_organism_query().order_by(Organism.internal_id)
```

Comment:
Alright, I agree!

It could be useful to separate the sqlalchemy dependency to a single layer in the application to make it easier to replace and not have our models depend on it directly. But that would involve more work than the current refactoring, for instance using a repository pattern and might not be worth it. Something we could consider in the future.